silver/models/payment_processors/fields.py

```diff
@@ -0,0 +1,43 @@
+from django.db.models import CharField
+from django.utils.encoding import smart_text
+
+from .managers import PaymentProcessorManager
+from .generics import GenericPaymentProcessor
+
+
+class PaymentProcessorField(CharField):
+    def __init__(self, *args, **kwargs):
+        max_length = 64
+        kwargs['max_length'] = max_length
+        super(PaymentProcessorField, self).__init__(*args, **kwargs)
+        self.validators = []
+
+    def get_internal_type(self):
+        return "CharField"
+
+    def from_string_value(self, value):
+        return PaymentProcessorManager.get(name=value) or value
+
+    def value_to_string(self, obj):
+        return smart_text(obj) if obj else u''
+
+    def from_db_value(self, value, *args, **kwargs):
+        return self.to_python(value)
+
+    def to_python(self, value):
+        if isinstance(value, GenericPaymentProcessor):
+            return value
+
+        if value is None:
+            return value
```

Comment:
True, but after the latest changes it won't anymore. "from_string_value" expects a string, not None.